File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/setmilestoneshapeproperties.rb
module Asposediagramjava
  module SetMilestoneShapeProperties
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shape_id = 1
        
        # Get timeline shape
        milestone = diagram.getPages().getPage("Flow 1").getShapes().getShape(shape_id)

        # Initialize MilestoneHelper object
        milestone_helper = Rjb::import('com.aspose.diagram.MilestoneHelper').new(milestone)
        
        # Set date format
        milestone_helper.setDateFormat(21)
        
        # Set auto update flag
        milestone_helper.setAutoUpdate(true)
        
        # Set milestone type
        milestone_helper.setType(6)

        # Save diagram
        diagram.save(data_dir + "SetMilestoneShapeProperties.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Set milestone shape properties."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/SetMilestoneShapeProperties.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\MilestoneHelper as MilestoneHelper;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class SetMilestoneShapeProperties{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        $shape_id=1;

        # Get timeline shape
        $milestone=$diagram->getPages()->getPage("Flow 1")->getShapes()->getShape($shape_id);

        # Initialize MilestoneHelper object
        $milestone_helper = new MilestoneHelper($milestone);

        # Set date format
        $milestone_helper->setDateFormat(21);

        # Set auto update flag
        $milestone_helper->setAutoUpdate(true);

        # Set milestone type
        $milestone_helper->setType(6);

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."SetMilestoneShapeProperties.vdx", $saveFileFormat->VDX);
        print "Set milestone shape properties.".PHP_EOL;
    }
}